Added EntityInteractObjective

joblo2213 opened the pull request.

Co0sh commented:
I'll test it tomorrow. It compiles, so I think it will be alright :3

joblo2213 commented:
I hope you don't forget about this^^

Co0sh commented:
I remember, I just haven't been working on BetonQuest since then :( This will be thefirst thing I do once I start again.

Co0sh commented on src/main/java/pl/betoncraft/betonquest/objectives/EntityInteractObjective.java:
```diff
@@ -0,0 +1,248 @@
+
+		public EntityInteractData(String instruction, String playerID, String objID) {
+			super(instruction, playerID, objID);
+			String[] args = instruction.split("-");
```
Here's a problem. UUIDs look like this: `718a5bd7-90cf-4e01-be6f-792d84e1b7a0`
There are `-` characters there, so splitting the instruction with them will yield a lot of cut UUIDs.

I suggest to use space as a separation of the instruction. That way `[0]` index will always be the amount left, and all other elements of that array will be the UUIDs of entities.

joblo2213 replied:
I think I wanted to do it like that (the UUIDs actually are seprated by a space, only between the integer and the uuids is a '-' ), but then somehow forgot about it while coding and messed it up^^...
I'll fix it tomorrow.

Co0sh commented on src/main/java/pl/betoncraft/betonquest/objectives/EntityInteractObjective.java:
```diff
@@ -0,0 +1,248 @@
+				completeObjective(playerID);
+			} else if (notify) {
+				// send a notification
+				Config.sendMessage(playerID, "mobs_to_kill", new String[] { String.valueOf(playerData.getAmount()) });
```
You can change the message used here to something like `mobs_to_click`. Just remember to add the English text of that message to the _messages.yml_ file. I can add Polish translation, and there are other people who can help us with other languages.

joblo2213 replied:
Tomorrow I'll implement it. I'll also add a german translation and maybe a french one (haven't spoke french for a long time but for this it should be enought^^)

joblo2213 commented:
I made the requested changes, added english, french and german messages and also added an additional cancel argument, so that the interactions can also be canceled. 
Hope now everything is fine
